app/src/main/java/edu/uc/groupproject/covid19tracker/service/CasesByCountryDataProvider.kt

```diff
@@ -25,7 +26,9 @@ class CasesByCountryDataProvider {
         val data = ArrayList<String>()
 
         client.newCall(request).enqueue(object : Callback {
-            override fun onFailure(call: Call, e: IOException) {}
+            override fun onFailure(call: Call, e: IOException) {
+                Log.d("onFailure", "onFailure from getCasesByCountryData called : ${e.printStackTrace()}")
+            }
```

Comment:
This comment is for all Log.d() statements. These may make it easier to debug in the future if any errors occur. https://developer.android.com/reference/kotlin/android/util/Log

Reply:
Excellent.  Many people leave the onFailure() empty.  Logging here is crucial.  If something goes wrong, the log statement will give you details!